docs: use proper documentation infra (#858)
grafana/grafana-image-renderer: docs/sources/_index.md

---
aliases:
  - ../administration/image_rendering/
  - ../image-rendering/
description: Image rendering
keywords:
  - grafana
  - image
  - rendering
  - plugin
labels:
  products:
    - enterprise
    - oss
title: Set up image rendering
weight: 1000
---

# Set up image rendering

The [Grafana Image Renderer service][github] exists to allow you to render
web pages in Grafana to PDFs, PNGs, and CSVs. For example, you can export your
panels or dashboards to PDFs to share with business stakeholders, or set up

> automatic exports for Slack alert notifications, such that you see your failing

Reviewer: ```suggestion
automatic exports for Slack alert notifications so you see your failing
```

alerts with context. Some features are exclusive to Grafana Enterprise and
Grafana Cloud.

{{< admonition type="caution" >}}
We historically provided this functionality as a Grafana plugin. This plugin is
deprecated and does not receive updates anymore. Similarly, we only support the
latest version of the service.
{{< /admonition >}}

{{< admonition type="note" >}}
At any given time, we commit to supporting _all_ stable currently supported
versions of Grafana. If you find a bug, consider reporting it on our [issue
tracker][issues].

[issues]: https://github.com/grafana/grafana-image-renderer/issues

{{< /admonition >}}

## Installation

To install the service, we recommend using Docker or other containerisation
software. This is not strictly required, however, so you can adapt to your
specific environment as needed.

We [ship images][image] for `linux/amd64` and `linux/arm64`; Windows and macOS
can run these with Docker Desktop as well. Additionally, we ship binaries for
Windows and Linux on our [GitHub Releases page][releases].

> To run the service, prefer to use a server with significant memory and CPU

Reviewer: ```suggestion
To run the service, you should use a server with significant memory and CPU
```

Reviewer: I'm just trying to be imperative in our wording here - "you should" could also sound discouraging fwiw

variable `GOMEMLIMIT` to a lower value than the limit, such as `1GiB`. You
should not aim for the `GOMEMLIMIT` to match the container's limit: Chromium
needs free memory on top. We recommend 1 GiB of `GOMEMLIMIT` per 8 GiB of
container memory limit.

If you do not use memory limits, the above configuration can still prove useful,
especially if you run the service alongside other services on the same host.

### Install with binaries

Binaries are released for Linux and Windows, for `amd64` and `arm64`
architectures. macOS is not supported at this time; you should prefer Docker
Desktop.

The binaries do not ship with a browser; you must install your own
Chromium-based browser on your own, and configure it to be used. Examples of
these include [Google Chrome](https://www.google.com/chrome/) and
[Microsoft Edge](https://www.microsoft.com/en-us/edge). Configure the path to
this browser binary with `--browser.path`.

The binaries are hosted on [GitHub Releases][releases]. Download the appropriate
binary for your platform, and run it from the terminal. You would run something
like `./grafana-image-renderer server`; see `--help` for more information.

> ### Configuration

Reviewer: I'd change this to an H2 to be on the same level as installation

The service can be configured via several paths:

- Set CLI flags. See `--help` for the available flag names.
- Use environment variables. See `--help` for the names and current values.
  Most, but not all, variables map 1:1 to the CLI flag names.
- Use a JSON or YAML configuration file. This must be in the service's current
  working directory, and must be named one of `config.json`, `config.yaml`, or
  `config.yml`. Dot-separated keys are nested keys. For example: `a.b` becomes
  `{"a": {"b": "VALUE"}}` in the file.

The current configuration options can all be accessed by checking `--help`.

As an example, see: `docker run --rm grafana/grafana-image-renderer:latest
server --help`.

### Security

The service requires a secret token to be present in all render requests. This
token is set by `--server.auth-token` (`AUTH_TOKEN`); you can specify multiple
and have a unique key per Grafana instance. By default, the token used is `-`.
In Grafana, you must set this to match one of the tokens with the `[rendering]

> renderer_token` (`GF_RENDERING_RENDERER_TOKEN`) setting.

Reviewer: ```suggestion
renderer_token` (`GF_RENDERING_RENDERER_TOKEN`) configuration setting.
```
